### PR TITLE
V3 fix sync call timeout

### PR DESCRIPTION
**Pull request by rekby**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

## Pull request type

Please check the type of change your PR introduces:

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?
Rare lost messages

## What is the new behavior?
Implement safe timeouts while call async code from sync

### CODE REVIEW

File: ydb/_topic_common/common.py [hunk 1]
```diff
@@ -60,3 +60,90 @@ def start_event_loop():
...
+            try:
+                res.set_result(await coro)
+            except asyncio.CancelledError:
+                res.set_exception(TimeoutError())
+
+        async def sleep0():
+            await asyncio.sleep(0)
+
+        coro_future = asyncio.run_coroutine_threadsafe(call_coro(), self._loop)
+        asyncio.run_coroutine_threadsafe(sleep0(), self._loop).result()
```
[the remark this answers is not in the document]
Comment (Valeria1235) [reply ydb/_topic_common/common.py:133]: I didn't understand what is problem this line fixes?

**Review by Valeria1235 — CHANGES_REQUESTED** [review]

File: ydb/_topic_common/common.py [hunk 2]
```diff
@@ -60,3 +60,90 @@ def start_event_loop():
...
+        """
+        f = self.unsafe_call_with_future(coro)
+        try:
+            return f.result(timeout)
+        except concurrent.futures.TimeoutError:
+            raise TimeoutError()
+        finally:
+            f.cancel()
+
+    def safe_call_with_result(self, coro: typing.Coroutine, timeout: TimeoutType):
```
[the remark this answers is not in the document]
Comment (rekby) [reply ydb/_topic_common/common.py:91]: It is easy to make error and forgot optional parameter. And at practice in most places timeout presented.

Optional timeout increase mistake error, but not change most of real code.

File: ydb/_topic_reader/topic_reader_sync.py [hunk 3]
```diff
@@ -150,37 +137,53 @@ def receive_batch(
...
         (for example if lost connection - commits will not re-send and committed messages will receive again)
         """
-        self._call_sync(self._async_reader.commit(mess), None)
+        self._check_closed()
+
+        self._caller.call_sync(self._async_reader.commit(mess))
 
     def commit_with_ack(
-        self, mess: ICommittable
+        self, mess: ICommittable, timeout: TimeoutType = None
```
[the remark this answers is not in the document]
Comment (rekby) [reply ydb/_topic_reader/topic_reader_sync.py:163]: It is internal method and ICommittable is real needs and read nned there better.

type union for public interface need only for better understand available types.

File: ydb/_topic_common/common.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Valeria1235) [reply ydb/_topic_common/common.py:92]: maybe there is some truth, but passing explicit None in the functions looks ugly and needlessly for my opinion.

**Review by Valeria1235 — APPROVED** [review]

**Pull request merged**